tests/integration/conftest.py

```
@@ -0,0 +1,314 @@
+    while time.time() - start < timeout:
+        if _is_port_open(host, port):
+            return True
+        time.sleep(interval)
+    return False
+
+
+def _is_docker_available() -> bool:
+    """Check if Docker is available and running."""
+    try:
+        result = subprocess.run(
+            ["docker", "info"],
```

Ruff (0.14.10)

Starting a process with a partial executable path

(S607)